display_list/display_list_unittests.cc

```diff
@@ -2116,6 +2116,25 @@ TEST(DisplayList, ClipRectAffectsClipBounds) {
   ASSERT_EQ(builder.getDestinationClipBounds(), initialDestinationBounds);
 }
 
+TEST(DisplayList, ClipRectAffectsClipBoundsWithMatrix) {
+  DisplayListBuilder builder;
+  SkRect clipBounds1 = SkRect::MakeLTRB(0, 0, 10, 10);
+  SkRect clipBounds2 = SkRect::MakeLTRB(10, 10, 20, 20);
+  builder.save();
+  builder.clipRect(clipBounds1, SkClipOp::kIntersect, false);
+  builder.translate(10, 0);
+  builder.clipRect(clipBounds1, SkClipOp::kIntersect, false);
+  ASSERT_EQ(builder.getDestinationClipBounds(), SkRect::MakeEmpty());
```

Comment:
Perhaps we should relax these to "get...Bounds().isEmpty()" rather than force a specific empty return value? Are they documented to return exactly that rectangle for an empty condition? I suppose we could just modify the test if we ever return a different value, but an implementation might decide to just max/min the 8 values and let their natural "emptiness" be a return value. We only get a hard-coded 0,0,0,0 rect because we use Skia for the intersections and they decided to return a false boolean with no changes to the rectangle for an empty intersection and so we have to manually force the answer to an empty answer (using MakeEmpty() for convenience).

Reply:
Done.